src/helpers/reorderHelper.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * items[beforeIndex]を基準に、selectedItemsをafterIndexの位置に移動する。
+ *
+ * 以下の場合は例外を投げる：
+ * - items内の要素が重複している
+ * - selectedItems内にitemsに存在しない要素がある
+ * - beforeIndexやafterIndexが範囲外
+ * - items[beforeIndex]がselectedItemsに含まれていない
+ */
+export function reorder<T>(
```

Comment:
これちょっとどういう想定でどう使うものなのかわかりにくいかもですね！
汎用的な並び替え関数を探してる人がこれを見たときにしばらくにらめっこすることになりそう。

ドラッグアンドドロップ用のものだとわかるものにしても良いかも。
こういう感じとかで。


```suggestion
/**
 * ドラッグ＆ドロップ操作において、選択された要素群を指定位置に移動する。
 *
 * 以下の場合は例外を投げる：
 * - items内の要素が重複している
 * - selectedItems内にitemsに存在しない要素がある
 * - beforeIndexやafterIndexが範囲外
 * - items[beforeIndex]がselectedItemsに含まれていない
 */
export function reorder<T>(
```

関数名もdragAndDropReorderみたいな感じでわかりやすくしても良いかも。

Reply:
ここはちょっとわかりやすい形にしときたみ。たぶん勘違いするので